openverse_catalog/dags/providers/provider_api_scripts/inaturalist.py

```diff
@@ -144,6 +143,73 @@ def compare_update_dates(
         raise AirflowSkipException("Nothing new to ingest")
 
     @staticmethod
+    def load_catalog_of_life_names():
+        COL_URL = "https://api.checklistbank.org/dataset/9840/export.zip?format=ColDP"
+        # Tried importing common.loader.paths.STAGING_DIRECTORY but it didn't work in
+        # local environment, TO DO: find a better place for this process.
+        DATA_DIR = Path(__file__).parents[4]
+        local_zip_file = "COL_archive.zip"
+        name_usage_file = "NameUsage.tsv"
+        vernacular_file = "VernacularName.tsv"
+        # download zip file from Catalog of Life
+        if (DATA_DIR / local_zip_file).exists():
+            logger.info(
+                f"No download from Catalog of Life. {DATA_DIR}/{local_zip_file} exists."
+            )
+        else:
+            with requests.get(COL_URL, stream=True) as r:
```

Comment:
Ooo, this is a new request pattern! Unfortunately we don't have a way to handle streaming yet in the [`DelayedRequester` machinery](https://github.com/WordPress/openverse-catalog/blob/c1b970b11dbf63a45a8c2d6a33d6058a662b7ac9/openverse_catalog/dags/common/requester.py#L103). Something to think about down the line maybe!

Reply:
Totally! I thought about trying to use `DelayedRequester`, and I think it does take `kwargs`, so technically it could just pass along the `stream=True`, but then it wouldn't have the rest of the pattern. It seems like it's more designed for a bunch of smaller requests than the one big one, but maybe I should at least be more explicit about retries? Or maybe it would be better to break this into smaller tasks and let airflow handle it?

Reply:
I think this is solid as is, because even with `DelayedRequester`'s retries we'd still need to change a LOT in order to get the context manager + `stream` to work. Something to look at again if it's a pattern we notice though!

Reply:
It looks like this would be possible _outside_ the context manager, but we would need to ensure the connection is closed once the operation is complete:
```python
r = None
try:
    r = self.delayed_requester.get(COL_URL, stream=True)
    with open(DATA_DIR / local_zip_file, "wb") as f:
        for chunk in r.iter_content(chunk_size=8192):
            f.write(chunk)
finally:
    if r is not None:
        r.close()
```